overlord/assertstate/validation_set_tracking.go

```diff
@@ -0,0 +1,138 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package assertstate
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+// ValidationMode reflects the mode of respective validation set, which is
+// either monitoring or enforcing.
+type ValidationMode int
+
+const (
+	Monitor ValidationMode = iota
+	Enforce
+)
+
+// ValidationTrackingKey is a key for tracking the given validation set.
+type ValidationTrackingKey struct {
+	AccoundID string
+	Name      string
+}
+
+func (v *ValidationTrackingKey) String() string {
```

Comment:
we would still need this but maybe as helper function ValidationSetKey(accountID, name) string